photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -160,6 +171,28 @@ final class Driver(val sc: SparkContext, val params: GameParams, implicit val lo
     }
 
   /**
+   * Run various checks on data (training or validation. Throws a RuntimeException if any of the checks does not pass.
+   *
+   * @note this is somewhat expensive, so we can turn it off
+   *
+   * Checks performed
+   * ================
+   * - all weights >= 0.0 and at least 1 weight > 0
```

Comment:
So just `at least 1 weight > 0` - that's what it decomposes into, unless we were explicitly checking for weights < 0